src/Core/src/Hosting/HandlerMauiAppBuilderExtensions.cs

```diff
@@ -10,12 +10,19 @@ public static class HandlerMauiAppBuilderExtensions
 	{
 		public static MauiAppBuilder ConfigureMauiHandlers(this MauiAppBuilder builder, Action<IMauiHandlersCollection>? configureDelegate)
 		{
-			builder.Services.TryAddSingleton<IMauiHandlersFactory, MauiHandlersFactory>();
+			ConfigureMauiHandlers(builder.Services, configureDelegate);
+			return builder;
+		}
+
+		public static IServiceCollection ConfigureMauiHandlers(this IServiceCollection services, Action<IMauiHandlersCollection>? configureDelegate)
```

Comment:
@PureWeen FYI we're adding an overload here for handler registration that goes off of IServiceCollection because in some Blazor scenarios that's all we have access to.

Reply:
Good catch, I forgot to point it out.